packages/inventory/src/containers/ServicePointProfile/tests/index.test.tsx

```diff
@@ -110,9 +110,8 @@ describe('Profile view Page', () => {
 
   it('renders when data is present', async () => {
```

Comment:
I feel there is so much we are not testing with just the inline snapshots? e.g Have tested the page title is correct? What if someone changes the classname of the div element from `inventory-profile-header` to `inventory-profile`, will the tests still pass? What about if I change a `Col` size?

Reply:
What if someone changes the `Button` from primary to secondary or the button size? Will the tests still hold?

Reply:
hmmm,.. 🤔

Reply:
As a rule of thumb I think we should try and avoid implementation details. which to me leads to harder question of what to consider as implementation detail.

- I do not have fast hard rules on what places one should add snapshot testing, if I had gun to my head, I would probably say snapshot testing is handy where a change in the ui could change/break how some other part of the page or app works
- However its still important that one indetifies this UI changes  in their smallest scope possible, in order to avoid large snapshots.
- In some other places, having a snapshot for instance to help identify changes in classNames or component structure adds no value to the tests. for instance; in certain cases, a change in a className in the snapshot, is not confirmation that the UI changed to look like what we want, to me this is valueless and thus I do not think its worth testing.
-  I also think, it boils down to what you want to check as part of the test, and what kind of meaningful assertions out testing tools allows us to do.

This probably does not give the answer to your questions @kelvin-muchiri , its just my 2 cents. I wonder what the rest of the team thinks @KipSigei @HaseemUlHaq @ashfahan @moshthepitt

Reply:
First, some links:

- The [original intro to jest snapshots by facebook](https://jestjs.io/blog/2016/07/27/jest-14.html)
- [Effective snapshot testing](https://kentcdodds.com/blog/effective-snapshot-testing) (I have shared this with all of you a bunch)
- [When to use jest snapshots](https://selleo.com/blog/when-to-use-jest-snapshots)

What are snapshot tests for anyway?  For me, they are a way to get some reasonable and cheap (as in easy to get) guarantees on the output being displayed to the user.

For example, if it is absolutely critical that there should be a green button displayed that has the words "buy now" then a snapshot test is ideal for this. i.e. that there is a button, and it is green, and it says "buy now".  Fun fact: something like this exists as a Reveal web app test.

Anyway for me this makes the "spirit" of snapshot testing clear i.e. to test things that are output to the screen

But everything else about snapshot testing is subjective:

- some people think huge snapshots are ok (I don't)
- sometimes you want to include things like class names, other times you don't
- etc etc

But if you agree with the purpose of this kind of testing, and you are a thoughtful programmer then it'll be okay in the end.  Otherwise your team-mate will bring it up in a review :)

Reply:
cool, always a nuanced topic, Thanks @kelvin-muchiri  for bringing this up, thanks @moshthepitt for your input, clear as always. I will  be picking up work on requested changes since Haseem has since been on leave.

Reply:
@p-netm Since the majority of the UI elements are Antd components, we should only be interested in the props passed to these components, which means we are not interested in the implementation. This is with regard to the [Effective snapshot testing ](https://kentcdodds.com/blog/effective-snapshot-testing) and similar articles. 

So if your component has other components as children, you should only be checking the props of these components at least. This is different from checking the implementation. This the rule of thumb I learnt from these articles

Also, if you can achieve it without using a snapshot, the better or with as minimal snapshots as possible. But I always make sure my `render` test cases asserts that the **order of elements is correct,** such that if I re-arrange the order of elements, the test should fail, and that the elements themselves are tested i.e through checking the props of a snapshot

Reply:
Since element props snapshots aren't that large for some elements It would be best to have maybe one test case with the props snapshots. This will help track when a dev changes the className, Id or the element tag.

Reply:
I kind of disagree on using snapshots for props, it seems like one of those cases where an explicit assertion works better. That being said,  I also feel iffy about testing props passed, to me that seems like an implementation detail.  tests should answer a question of whether a certain business requirement has been be met. I do not think testing that supplied props looking a certain way correlates with encoding business requirements in the test. IMO it's an unavoidable technical detail towards satisfying the requirement, but isn't an affirmation that the requirement is met in itself.

of-course, this does not apply everywhere and its subjective according to whoever is writing the tests.

seems like we should have mini tech meets to talk this stuff.

Reply:
> I kind of disagree on using snapshots for props, it seems like one of those cases where an explicit assertion works better. That being said, I also feel iffy about testing props passed, to me that seems like an implementation detail. tests should answer a question of whether a certain business requirement has been be met. I do not think testing that supplied props looking a certain way correlates with encoding business requirements in the test. IMO it's an unavoidable technical detail towards satisfying the requirement, but isn't an affirmation that the requirement is met in itself.

I think sometimes you want to assert that a component received the correct props, especially when those props change how the component works. e.g. is some table component has a prop to control whether table cell borders or not, and you know that you absolutely need the table to not have borders then it might make sense to encode this in a test that would break if the correct prop was not passed in.  

But obviously every case is different.  The important thing is to have a clear idea what you are trying to test and why.